crates/shared/src/order_validation.rs

```diff
@@ -459,6 +467,22 @@ impl OrderValidating for OrderValidator {
             return Err(ValidationError::ZeroAmount);
         }
 
+        // For now the field is optional. If it exists, we validate it. Eventually
+        // the field wil be mandatory. Or we will default it to the empty string, which
+        // maps to all zero app data hash. That offers some degree of backward
+        // compatibility.
+        if let Some(full_app_data) = &order.full_app_data {
+            let validated = self
+                .app_data_validator
+                .validate(full_app_data.as_bytes())
+                .map_err(ValidationError::InvalidAppData)?;
+            if validated.hash != order.data.app_data {
+                return Err(ValidationError::AppDataHashMismatch {
+                    actual: validated.hash,
+                });
```

Comment:
An API user already knows that value because they sent the request. But I guess it could clear up some confusion when they didn't know about the app data field at all. I'll add it.